### PR TITLE
PYTHON-3381 Improve readability of sphinx docs

### PR DESCRIPTION
**Pull request by blink1073**
image

image

### CODE REVIEW

File: bson/json_util.py [hunk 1]
```diff
@@ -372,6 +372,7 @@ def with_options(self, **kwargs: Any) -> "JSONOptions":
             >>> json_options.tz_aware
             False
 
+
```
Comment (juliusgeo) [inline bson/json_util.py:375]: Here and above seem to be accidental changes.

Comment (blink1073) [reply bson/json_util.py:375]: I needed them to avoid sphinx warnings about indentation.

**Comment by ShaneHarvey** [comment]
One problem I noticed is that multiple classes show up twice in the docs now. For example:
- https://pymongo--1156.org.readthedocs.build/en/1156/api/bson/binary.html#bson.binary.Binary
- https://pymongo--1156.org.readthedocs.build/en/1156/api/bson/index.html#bson.Binary

**Comment by ShaneHarvey** [comment]
It looks like the typehints do not show up in the method signature or in the Parameter list: https://pymongo--1156.org.readthedocs.build/en/1156/api/bson/index.html#bson.BSON.encode 

Why is that?

**Comment by blink1073** [comment]
`Binary` was already duplicated in the docs, since it is re-exported.

**Comment by blink1073** [comment]
So perhaps we should not use [sphinx-autodoc-typehints](https://github.com/tox-dev/sphinx-autodoc-typehints) unless we switch to a standard format.

**Comment by ShaneHarvey** [comment]
> Binary was already duplicated in the docs, since it is re-exported.

Could you open a bug for this? Looks like it was caused by #994.

> So perhaps we should not use [sphinx-autodoc-typehints](https://github.com/tox-dev/sphinx-autodoc-typehints) unless we switch to a standard format.

SGTM, let's do that in another ticket?

File: bson/json_util.py (hunk 1 above)
Comment (blink1073) [reply bson/json_util.py]: Removed now that we're not using `sphinx-autodoc-typehints`.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Love the new theme, thanks for finding it. Did you notice that you can toggle between light/dark themes in the top right?? I do still see a few problems though, like these two issues [on this page](https://pymongo--1156.org.readthedocs.build/en/1156/python3.html): 
Screen Shot 2023-02-15 at 12 00 07 PM

Could you take a look through the pages to see if you find anything else?

**Comment by blink1073** [comment]
I fixed those and a few other places I saw.  I went through all of the docs pages except for the secondary API docs pages.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**